src/communication.cpp

```diff
@@ -199,6 +207,23 @@ void Communication::play(const std::string& file)
     pub_play_.publish(str);
 }
 
+void Communication::sendPoseEstimate(const geometry_msgs::Transform& pose)
+{
+    // Publish tf transform
+    geometry_msgs::TransformStamped transformStamped;
+    transformStamped.header.stamp = ros::Time::now();
+    transformStamped.header.frame_id = "map";
+    transformStamped.child_frame_id = "internal/base_link";
+    transformStamped.transform = pose;
+    pub_tf2->sendTransform(transformStamped);
+}
+
+bool Communication::getMapConfig(MapConfig& config) {
+    if (!mapconfig.mapInitialised) {mapdata_cb_queue_.callAvailable();}//try to initialise the map
```

Comment:
Please don't use these inline ifs
```suggestion
    if (!mapconfig.mapInitialised)
        mapdata_cb_queue_.callAvailable();  // Try to initialise the map
```